modules/sharedaddy/recaptcha.php

```diff
@@ -0,0 +1,179 @@
+				'tabindex' => 0,
+			),
+		);
+	}
+
+	/**
+	 * Calls the reCAPTCHA siteverify API to verify whether the user passes
+	 * CAPTCHA test.
+	 *
+	 * @param string $response  The value of 'g-recaptcha-response' in the submitted
+	 *                          form.
+	 * @param string $remote_ip The end user's IP address.
+	 *
+	 * @return bool|WP_Error Returns true if verified. Otherwise WP_Error is returned.
+	 */
+	public function verify( $response, $remote_ip ) {
+		// No need make a request if response is empty.
+		if ( empty( $response ) ) {
+			return new WP_Error( 'missing-input-response', $this->error_codes['missing-input-response'], 400 );
+		}
+
+		$resp = wp_remote_post( self::VERIFY_URL, $this->get_verify_request_params( $response, $remote_ip ) );
+		if ( is_wp_error( $resp ) ) {
+			return $resp;
+		}
+
+		$resp_decoded = json_decode( wp_remote_retrieve_body( $resp ), true );
+		if ( ! $resp_decoded ) {
+			return new WP_Error( 'invalid-json', $this->error_codes['invalid-json'], 400 );
+		}
+
+		// Default error code and message.
+		$error_code    = 'unexpected-response';
+		$error_message = $this->error_codes['unexpected-response'];
+
+		// Use the first error code if exists.
+		if ( isset( $resp_decoded['error-codes'] ) && is_array( $resp_decoded['error-codes'] ) ) {
+			if ( isset( $resp_decoded['error-codes'][0] ) && isset( $this->error_codes[ $resp_decoded['error-codes'][0] ] ) ) {
+				$error_message = $this->error_codes[ $resp_decoded['error-codes'][0] ];
+				$error_code    = $resp_decoded['error-codes'][0];
+			}
+		}
+
+		if ( ! isset( $resp_decoded['success'] ) ) {
+			return new WP_Error( $error_code, $error_message );
+		}
+
+		if ( true !== $resp_decoded['success'] ) {
+			return new WP_Error( $error_code, $error_message );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get siteverify request parameters.
+	 *
+	 * @param string $response  The value of 'g-recaptcha-response' in the submitted
+	 *                          form.
+	 * @param string $remote_ip The end user's IP address.
+	 *
+	 * @return array
+	 */
+	public function get_verify_request_params( $response, $remote_ip ) {
+		return array(
+			'body' => array(
+				'secret'   => $this->secret_key,
+				'response' => $response,
+				'remoteip' => $remote_ip,
+			),
+			'sslverify' => true,
+		);
+	}
+
+	/**
+	 * Get reCAPTCHA HTML to render.
+	 *
+	 * @return string
+	 */
+	public function get_recaptcha_html() {
+		return sprintf(
+			'
+			<div
+				class="%s"
+				data-sitekey="%s"
+				data-theme="%s"
+				data-type="%s"
+				data-tabindex="%s"></div>
+			<script type="text/javascript" src="https://www.google.com/recaptcha/api.js?hl=%s"%s></script>
+			',
+			esc_attr( $this->config['tag_class'] ),
+			esc_attr( $this->site_key ),
+			esc_attr( $this->config['tag_attributes']['theme'] ),
+			esc_attr( $this->config['tag_attributes']['type'] ),
+			esc_attr( $this->config['tag_attributes']['tabindex'] ),
+			esc_attr( $this->config['language'] ),
```

Comment:
It would be better to use `rawurlencode` instead of `esc_attr` since it's used as a URL param.